### PR TITLE
Old PR / to be splitted 

### PR DESCRIPTION
**Pull request by alucas**

### CODE REVIEW

File: src/withPayline.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import React from 'react';
+import React, { PropsWithChildren } from 'react';
 import PaylineProvider from './PaylineProvider';
 
-const withPayline = (production: boolean) => (WrappedComponent: React.ComponentType) => (props: any) => (
+const withPayline = (production: boolean) => <TProps,>(WrappedComponent: React.ComponentType<TProps>) => (props: PropsWithChildren<TProps>) => (
```
Comment (alex-pex) [inline src/withPayline.tsx:4]: Why the trailing comma? Is it a typo?

**Review by alex-pex — CHANGES_REQUESTED** [review]
Thank you for your contribution.
I need to do some tests to ensure the type changes don't break something in our app.

Do you mind splitting your PR in two, to seperate the type changes from the code and doc refactoring?

File: src/withPayline.tsx (hunk 1 above)
Comment (alucas) [reply src/withPayline.tsx:4]: Hi, it's not a typo, it's quirk of Typescript with JSX

Comment (alucas) [reply src/withPayline.tsx:4]: I will split the PR asap

**Comment by alex-pex** [comment]
(previous comment erased, it was a mistake)

I did some cherry picking and manual fixes. I will put more details in the child PRs.

**Pull request closed without merging**